Sources/SwiftMemcache/MemcachedResponseDecoder.swift

```diff
@@ -115,40 +114,63 @@ struct MemcachedResponseDecoder: NIOSingleStepByteToMessageDecoder {
             }
 
             let returnCode = MemcachedResponse.ReturnCode(bytes)
-            self.nextStep = .dataLengthOrFlag(returnCode)
+            self.nextStep = .dataLength(returnCode)
             return .continueDecodeLoop
 
-        case .dataLengthOrFlag(let returnCode):
+        case .dataLength(let returnCode):
             if returnCode == .VA {
-                // TODO: Implement decoding of data length
-                fatalError("Decoding for VA return code is not yet implemented")
-            }
+                // Advance to the first non-whitespace character
+                while let currentByte = buffer.getInteger(at: buffer.readerIndex, as: UInt8.self), currentByte == UInt8.whitespace {
+                    buffer.moveReaderIndex(forwardBy: 1)
+                }
 
-            guard let nextByte = buffer.readInteger(as: UInt8.self) else {
-                return .waitForMoreBytes
+                guard let dataLength = buffer.readIntegerFromASCII() else {
+                    throw MemcachedDecoderError.unexpectedCharacter(buffer.readableBytesView[buffer.readerIndex])
+                }
+
+                // Skip over the whitespace, newline or carriage return
+                while let currentByte = buffer.getInteger(at: buffer.readerIndex, as: UInt8.self),
+                      [UInt8.whitespace, UInt8.newline, UInt8.carriageReturn].contains(currentByte) {
+                    buffer.moveReaderIndex(forwardBy: 1)
+                }
+
+                let isNextByteWhitespace = buffer.getInteger(at: buffer.readerIndex) == UInt8.whitespace
+                self.nextStep = isNextByteWhitespace ? .flags(returnCode, dataLength) : .decodeValue(returnCode, dataLength, nil)
+                return .continueDecodeLoop
+            } else {
+                self.nextStep = .flags(returnCode, nil)
+                return .continueDecodeLoop
             }
 
-            if nextByte == UInt8.whitespace {
-                self.nextStep = .decodeNextFlag(returnCode, nil)
+        case .flags(let returnCode, let dataLength):
+            let flags = buffer.readMemcachedFlags()
```

Comment:
So this is where it gets a bit tricky and I would like us to take a shortcut for now. The problem is that we can have a partial buffer where we don't have the carriageReturn&newline yet and we would need to support reading partial flags. This also includes reading the partial tokens of flags which might be quite big. In a best world, we would go flag by flag and move ourselves forward. However, implementing this requires a bit of logic. What I would do for now is trying to scan the buffer forward to see if we have a `\r\n` in the buffer. If we do we are going to try and decode if not we are going to return `needMoreData` and let's create an issue for us to revisit this in the future.

Reply:
Sounds good, we will create a new issue to handle the partial buffer implementation.